### PR TITLE
feat(roles): Return access/scopes on Projects

### PR DESCRIPTION
**Pull request by leedongwei**

### CODE REVIEW

File: src/sentry/api/serializers/models/project.py [hunk 1]
```diff
@@ -93,22 +93,44 @@ def get_access_by_project(
...
 
+    org_ids = {i.organization_id for i in projects}
+    all_org_roles = get_org_roles(org_ids, user)
     is_superuser = request and is_active_superuser(request) and request.user == user
+    prefetch_related_objects(projects, "organization")
+
     result = {}
     for project in projects:
-        is_member = any(t.id in team_memberships for t in project_team_map.get(project.id, []))
+        parent_teams = [t.id for t in project_team_map.get(project.id, [])]
```
Comment (nhsiehgit) [inline src/sentry/api/serializers/models/project.py:105]: what's a parent team?

can we have teams within teams??

Comment (leedongwei) [reply src/sentry/api/serializers/models/project.py:106]: A project can be a child of several teams.

File: src/sentry/api/serializers/models/project.py [hunk 2]
```diff
@@ -93,22 +94,44 @@ def get_access_by_project(
...
             or project.organization.flags.allow_joinleave
             or any(roles.get(org_role).is_global for org_role in org_roles)
         )
-        result[project] = {"is_member": is_member, "has_access": has_access}
+
+        team_scopes = set()
+        if has_access:
+            # Project can be the child of several Teams, and the User can join
+            # several Teams and receive roles at each of them,
+            team_scopes = set().union(*[m.get_scopes() for m in member_teams])
```
Comment (cathteng) [inline src/sentry/api/serializers/models/project.py:122]:
```suggestion
            team_scopes = team_scopes.union(*[m.get_scopes() for m in member_teams])
```

**Review by cathteng — APPROVED** [review]
logic lgtm

**Pull request merged**